code/modules/mining/smelter.dm

```diff
@@ -0,0 +1,209 @@
+/obj/machinery/smelter
+	name = "smelter"
+	desc = "An old Sendarian tool."
+	icon = 'icons/obj/blacksmithing.dmi'
+	icon_state = "forge"
+	density = TRUE
+	anchored = FALSE
+
+/obj/machinery/smelter/attackby(obj/item/weapon/W, mob/user, params)
+	var/smelting_result = W.on_smelt()
+	if(!smelting_result)
+		return ..()
+	if(user.drop_item())
+		to_chat(user, "You smelt [W].")
+		qdel(W)
+		var/obj/item/weapon/reagent_containers/glass/bucket/AB = new(get_turf(src))
+		AB.reagents.add_reagent(smelting_result, 75)
+		AB.reagents.chem_temp = 1000
+		AB.reagents.handle_reactions()
+		AB.name = "bucket of [AB.reagents.get_master_reagent_name()]"
+
+
+/obj/machinery/anvil
+	name = "anvil"
+	desc = "Goodman Durnik, is that you?"
+	icon = 'icons/obj/blacksmithing.dmi'
+	icon_state = "anvil"
+	density = TRUE
+	anchored = FALSE
+	var/obj/item/weapon/reagent_containers/glass/mold/current_mold = null
+	var/mutable_appearance/my_mold = null
+
+/obj/machinery/anvil/attackby(obj/item/weapon/W, mob/user, params)
+	if(!current_mold && istype(W, /obj/item/weapon/reagent_containers/glass/mold))
+		var/obj/item/weapon/reagent_containers/glass/mold/M = W
+		var/datum/reagent/R = M.reagents.get_master_reagent()
+		if(R && R.volume == 25)
+			if(user.drop_item())
+				to_chat(user, "You place [M] on [src].")
+				M.loc = src
+				current_mold = M
+				my_mold = mutable_appearance('icons/obj/blacksmithing.dmi', M.icon_state)
+				add_overlay(my_mold)
+				return
+		if(R && R.volume)
+			to_chat(user, "There's not enough in the mold to make a full cast!")
+			return
+		else
+			to_chat(user, "There's nothing in the mold!")
+			return
+	if(istype(W, /obj/item/weapon/smith_hammer))
+		if(current_mold)
+			to_chat(user, "You break the result out of the mold.")
+			new current_mold.type(get_turf(src))
+			var/datum/reagent/R = current_mold.reagents.get_master_reagent()
+			var/obj/item/I
+			if(!istype(current_mold, /obj/item/weapon/reagent_containers/glass/mold/bar))
+				I = new current_mold.produce_type(get_turf(src))
+				I.smelted_material = new R.type()
+				I.post_smithing()
+			else
+				I = new R.produce_type(get_turf(src))
+			qdel(current_mold)
+			cut_overlay(my_mold)
+			my_mold = null
+			current_mold = null
+		else
+			to_chat(user, "There's nothing in the mold!")
+
+/obj/item/weapon/smith_hammer
+	name = "smith's hammer"
+	desc = "John was here."
```

Comment:
Seriously, though. The idea of it is good, and it makes a good enough reference on its own. You don't need stuff like this description to make things more confusing for people who don't get it.